src/commons/workspace/WorkspaceActions.ts

```diff
@@ -385,20 +385,29 @@ export const notifyProgramEvaluated = (
     workspaceLocation
   });
 
-export const toggleUsingSubst = (usingSubst: boolean, workspaceLocation: WorkspaceLocation) =>
-  action(TOGGLE_USING_SUBST, { usingSubst, workspaceLocation });
+// Only playground and sicp supports subst visualizer
+export const toggleUsingSubst = (
+  usingSubst: boolean,
+  workspaceLocation: Extract<WorkspaceLocation, 'playground' | 'sicp'>
```

Comment:
We can consider abstracting `'playground' | 'sicp'` as a type e.g. WorkspaceLocationsWithTools

Reply:
Okay, will do so.